forui/lib/src/widgets/accordion.dart

```diff
@@ -0,0 +1,525 @@
+import 'dart:math' as math;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:meta/meta.dart';
+
+import 'package:forui/forui.dart';
+import 'package:forui/src/foundation/tappable.dart';
+import 'package:forui/src/foundation/util.dart';
+
+/// A controller that stores the expanded state of an [_Item].
+interface class FAccordionController extends ChangeNotifier {
+  /// The duration of the expansion and collapse animations.
+  final Duration duration;
+  final List<(AnimationController, Animation)> _controllers;
+  final Set<int> _values;
+  final int _min;
+  final int? _max;
+
+  /// Creates a [FAccordionController].
+  ///
+  /// The [min] and [max] values are the minimum and maximum number of selections allowed. Defaults to no minimum or maximum.
+  ///
+  /// # Contract:
+  /// * Throws [AssertionError] if [min] < 0.
+  /// * Throws [AssertionError] if [max] < 0.
+  /// * Throws [AssertionError] if [min] > [max].
+  FAccordionController({
+    int min = 0,
+    int? max,
+    Set<int>? values,
+    this.duration = const Duration(milliseconds: 500),
+  })  : _min = min,
+        _max = max,
+        _controllers = [],
+        _values = values ?? {},
+        assert(min >= 0, 'The min value must be greater than or equal to 0.'),
+        assert(max == null || max >= 0, 'The max value must be greater than or equal to 0.'),
+        assert(max == null || min <= max, 'The max value must be greater than or equal to the min value.');
+
+  /// Adds an item to the accordion.
+  void addItem(int index, AnimationController controller, Animation expand) {
+    _controllers.add((controller, expand));
+    if (controller.value == 1.0) {
+      _values.add(index);
+    }
+  }
+
+  /// Removes an item from the accordion.
+  void removeItem(int index) {
+    _controllers.removeAt(index);
+    _values.remove(index);
+  }
+
+  /// Convenience method for toggling the current expanded status.
+  ///
+  /// This method should typically not be called while the widget tree is being rebuilt.
+  Future<void> toggle(int index) async {
+    if (_controllers[index].$2.value == 100) {
```

Comment:
Should it be 100? I thought it was normally 1.0 for animations.

Reply:
This can be either 100 when we track the `animation.value`, or 1.0 if we are tracking the `controller.value`.

Which do you prefer?